include/votca/xtp/bse_matrix_free.h

```diff
@@ -0,0 +1,171 @@
+/*
+ *            Copyright 2009-2018 The VOTCA Development Team
+ *                       (http://www.votca.org)
+ *
+ *      Licensed under the Apache License, Version 2.0 (the "License")
+ *
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *              http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#ifndef _VOTCA_XTP_BSE_MF_H
+#define _VOTCA_XTP_BSE_MF_H
+
+#include <votca/xtp/orbitals.h>
+#include <votca/xtp/rpa.h>
+#include <votca/xtp/threecenter.h>
+#include <votca/xtp/qmstate.h>
+
+#include <votca/xtp/matrixfreeoperator.h>
+
+namespace votca {
+namespace xtp {
+
+class BSE_MF : public MatrixFreeOperator {
```

Comment:
I do not get this class, I thought the idea was to use an independent class `BSEHamiltonian` and derive singlethamiltonian  and triplethamiltonian from it. via

`class singlethamiltonian: public BSEHamiltonian, public MatrixFreeOperator`